source/developers-guide/plugin-extension-by-plugin/index.md

```diff
@@ -0,0 +1,684 @@
+            <argument type="service" id="service_container" />
+            <tag name="shopware.event_subscriber" />
+        </service>
+
+    </services>
+
+</container>
+```
+
+### SwagExtendCustomProducts/SwagExtendCustomProducts.php
+is like the Bootstrap.php in the old plugin-system. 
+Overwrite functions to execute your code here.
+ 
+For more information about the functions you can use and overwrite take a look into the wiki <a href="{{ site.url }}/developers-guide/plugin-system/">The 5.2 Plugin system</a> or in the class _\Shopware\Components\Plugin_ which you can find in the folder _ /engine/Shopware/Components_.
+
+in our example we overwrite the build and the install methods. 
+ - The build method, to add a new parameter into the dependency container.
+Whenever you have the container for disposal you can call the parameter by:
+
+```php
+$this->container->getParameter('swag_extend_custom_products.plugin_dir');
+```
+ - The install method, to initialize our installation procedure.
+
+```php
+<?php
+
+namespace SwagExtendCustomProducts;
+
+use Shopware\Components\Plugin;
+use Shopware\Components\Plugin\Context\InstallContext;
+use SwagExtendCustomProducts\Setup\Installer;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class SwagExtendCustomProducts extends Plugin
+{
+    /**
+     * Add a new parameter to the container, to retrieve the plugin path at any time.
+     *
+     * @param ContainerBuilder $container
+     */
+    public function build(ContainerBuilder $container)
+    {
+        $container->setParameter('swag_extend_custom_products.plugin_dir', $this->getPath());
+
+        parent::build($container);
+    }
+
+    /**
+     * Overwrites the install method to execute specific installation code.
+     *  
+     * @param InstallContext $context
+     */
+    public function install(InstallContext $context)
+    {
+        $installer = new Installer($this->container);
+        $installer->install();
+    }
+}
+```
+
+### SwagExtendCustomProducts/Installer.php
+To keep track of the code, swap out the installation code in this extra file. 
+In this example it checks if the plugin "Custom Products (v2)" is installed, but you can add other code here if necessary for your installation.
+
+```php
+<?php
+
+namespace SwagExtendCustomProducts\Setup;
+
+use RuntimeException;
+use SwagExtendCustomProducts\Components\Verifier;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class Installer
+{
+    /** @var ContainerInterface */
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * @return bool
+     */
+    public function install()
+    {
+        // Checks if the plugin Custom Products (v2) is installed.
+        $verifier = new Verifier($this->container->get('dbal_connection'));
```

Comment:
Please move the verifier source code to this installer, so other can see how you check if the other plugin is installed/existing/activated.